src/simulators/matrix_product_state/matrix_product_state_internal.cpp

```diff
@@ -1473,14 +1473,16 @@ reg_t MPS::sample_measure_using_probabilities_internal(const rvector_t &rnds,
 }
 
 reg_t MPS::apply_measure(const reg_t &qubits, RngEngine &rng) {
-  // since input is always sorted in qasm_controller, therefore, we must return the qubits 
-  // to their original location (sorted)
+  // apply_measure_internal requires the qubits to be sorted
   move_all_qubits_to_sorted_ordering();
-  return apply_measure_internal(qubits, rng);
+  reg_t sorted_qubits = qubits;
+  std::sort(sorted_qubits.begin(), sorted_qubits.end());
+  return apply_measure_internal(sorted_qubits, rng);
 }
 
-reg_t MPS::apply_measure_internal(const reg_t &qubits, 
-				  RngEngine &rng) {
+// The caller to apply_measure_internal is responsible to call 
+// move_all_qubits_to_sorted_ordering before calling this function
+reg_t MPS::apply_measure_internal(const reg_t &sorted_qubits, RngEngine &rng) {
```

Comment:
Add a comment that `sorted_qubits` must be sorted

Reply:
Is it really expensive to verify that `sorted_qubits` is sorted, and that the qubits are sorted?

Reply:
I added a comment that `sorted_qubits` must be sorted.
Regarding adding a verification for the above two conditions, the problem is this will be called for every shot, so I think it is a waste to spend time on this.